h/services/bulk_api/lms_stats.py

```diff
@@ -71,85 +77,102 @@ def _annotation_query(self, groups: list[str]):
                 Group.authority_provided_id.in_(groups),
             )
         )
-
-    def get_counts_by_user(
-        self, groups: list[str], assignment_id: str
-    ) -> list[CountsByUser]:
-        """
-        Get basic stats per user for an LMS assignment.
-
-        :param groups: List of "authority_provided_id" to filter groups by.
-        :param assignment_id: ID of the assignment we are generating the stats for.
-        """
-        annos_query = (
-            self._annotation_query(groups)
-            .where(
+        if assignment_id:
```

Comment:
Moving all the filtering here. We decouple filtering for the rest of the query building.